utils_test.go

```diff
@@ -289,3 +343,35 @@ func ExampleWriteXML() {
 	//  <contents>dummy content</contents>
 	// </dummyXML>
 }
+
+func TestEnsureDir(t *testing.T) {
+	t.Run("test if directory gets created", func(t *testing.T) {
+		dirName := "tmp/ensure-dir-test"
+		defer func() {
+			err := os.RemoveAll(dirName)
+			if err != nil {
+				t.Error("Directory created by EnsureDir should be removable")
+			}
+		}()
+
+		err := EnsureDir(dirName)
+		if err != nil {
+			t.Errorf("EnsureDir should not return error when creating directory: %s", err)
+		}
+	})
+
+	t.Run("test catch errors", func(t *testing.T) {
+		errMessage := "Fail to create dirs"
+		monkey.Patch(os.MkdirAll, func(path string, perm os.FileMode) error {
```

Comment:
I'm not sure I would monkey patch here instead of trying to create a directory in some way that fails that you forsee the user doing. However, I think its fine for now!